apps/angular/crud/src/app/data-access/todos.store.ts

```diff
@@ -0,0 +1,125 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { randText } from '@ngneat/falso';
+import { ComponentStore } from '@ngrx/component-store';
+import { Todo } from '../model/todo.interface';
+
+export interface TodosState {
+  todos: Todo[];
+  loading: boolean;
+  error: string | null;
+  disabledTodosIds: number[];
+  errorTodosIds: number[];
+}
+
+@Injectable()
+export class TodosStore extends ComponentStore<TodosState> {
+  constructor(private http: HttpClient) {
+    super({
+      todos: [],
+      loading: false,
+      error: null,
+      disabledTodosIds: [],
+      errorTodosIds: [],
+    });
+  }
+
+  load() {
+    this.patchState({ loading: true, error: null });
```

Comment:
you should create a property that have both loading and error state. Less error prone

`callstate = Loading | {err: stirng}`

Reply:
changed to one property use